compiler/src/main/java/com/squareup/anvil/compiler/ModuleMergerIr.kt

```diff
@@ -297,7 +314,12 @@ internal class ModuleMergerIr(
         }
       }
 
-    declaration.annotations += annotationConstructorCall
+    declaration.addAnnotation(
+      annotationConstructorCall.toAnnotationReference(
+        pluginContext,
+        declaration
+      )
+    )
```

Comment:
Hm, interesting. Not sure if this is actually needed. Our `ClassReference` instances capture the current state of the compilation, we run our logic and then modify the existing code. After that we're done. We don't process or work with the modified code. I don't think you need to update class reference and you can just change the underlying IR structure.

Reply:
Do you plan to address this here?

Reply:
Giving it a shot now and will address here if it proves unnecessary